tasks/tests/unit/test_upload_task.py

```diff
@@ -609,8 +595,7 @@ def test_upload_task_call_multiple_processors(
                 ],
                 report_code=None,
                 in_parallel=False,
-                is_final=False,
-            ),
+                is_final=True,
```

Comment:
This is an interesting driveby fix. The previous code would get this wrong for the last chunk happens to be on a chunk border. I initially replicated the same buggy behavior, but then found a better fix for it that correctly flags the final chunk/task.